### PR TITLE
[SYCL][NFC] Detach unit-tests naming from UR

### PR DESCRIPTION
**Pull request by AlexeySachkov**
This is a follow-up from #14145. Our unit tests have mock classes corresponding to device images and other entries our compiler generates. They are all named using `Ur` prefix, but they have nothing to do with Unified Runtime, they are part of our compiler interface.

This patch renames associated files and classes so they do not refer to `Ur` anymore.
This is a mechanical find-and-replace change.

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
This PR will conflict with #14816. I have no strict preference about which one goes first, but I think that if we merge this one first, then I will have left conflicts to resolve.

**Comment by aelovikov-intel** [comment]
> then I will have left conflicts to resolve.

Is that s/left/less/  ?

**Comment by AlexeySachkov** [comment]
> > then I will have left conflicts to resolve.
> 
> Is that s/left/less/ ?

Yes

File: sycl/unittests/helpers/MockDeviceImage.hpp [hunk 1]
```diff
@@ -110,17 +112,17 @@ class UrOffloadEntry {
 };
 
 /// Generic array of UR entries.
-template <typename T> class UrArray {
+template <typename T> class Array {
```
[the remark this answers is not in the document]
Comment (AlexeySachkov) [reply sycl/unittests/helpers/MockDeviceImage.hpp:115]: The thing is that this wrapper not only used for entries, but also for properties and maybe for something else as well. So far I had no other ideas, but I will think about the name more.

[the remark this answers is not in the document]
Comment (AlexeySachkov) [reply sycl/unittests/helpers/MockDeviceImage.hpp:115]: What about `ArrayOf`? It is still generic, but it is not that close to `array`, so probably it better conveys that it is a special wrapper that we need

[the remark this answers is not in the document]
Comment (aelovikov-intel) [reply sycl/unittests/helpers/MockDeviceImage.hpp:115]: I'm more concerned by the fact that this wrapper has multiple fields, which is not clear from the name still...

[the remark this answers is not in the document]
Comment (AlexeySachkov) [reply sycl/unittests/helpers/MockDeviceImage.hpp:115]: What about just dropping `UrArray` completely? I have a POC for it in #15014. Another alternative would be to hide `UrArray` from unit-tests making it internal to mock objects. I.e. a unit-test won't use `UrArray` and will pass a regular `vector` to `UrPropertySet`, but under the hood that vector will be converted into `UrArray`

[the remark this answers is not in the document]
Comment (AlexeySachkov) [reply sycl/unittests/helpers/MockDeviceImage.hpp:115]: Another approach here (which I personally like the most) is to separately rename `UrArray` and "hide" it as an implementation detail: #15604.

@aelovikov-intel, @steffenlarsen, could you please take a look? I know that Steffen has no objections against #15014, but I wonder which approach do you folks think is better here?

[the remark this answers is not in the document]
Comment (aelovikov-intel) [reply sycl/unittests/helpers/MockDeviceImage.hpp:115]: I like https://github.com/intel/llvm/pull/15604 a lot.

**Review by aelovikov-intel — APPROVED** [review]

**Comment by AlexeySachkov** [comment]
3eafbcb1590f8c31017d47e52d59ed54944f380e - a very small change which I missed earlier because virtual functions unit-tests were disabled at the time.

**Review by aelovikov-intel — APPROVED** [review]

**Comment by AlexeySachkov** [comment]
I've verified locally that build and unit-tests still pass after merge with `sycl` branch, so I will proceed with this PR.

**Pull request merged**